modules/hotmaps/1.0/hotmaps.smk

```diff
@@ -0,0 +1,837 @@
+    _hotmaps_multiple_test_correct,
+    _hotmaps_find_gene,
+    _hotmaps_find_structure,
+    _hotmaps_output,
+    _hotmaps_detailed_hotspots,
+    _hotmaps_aggregate,
+    _hotmaps_all,
+
+##### RULES #####
+
+if "launch_date" in CFG:
+    launch_date = CFG["launch_date"]
+else:
+    launch_date = datetime.today().strftime('%Y-%m')
+
+# Interpret the absolute path to this script so it doesn't get interpreted relative to the module snakefile later.
+PREPARE_MAFS = os.path.abspath(config["lcr-modules"]["hotmaps"]["maf_processing"]["prepare_mafs"])
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _hotmaps_input_maf:
+    input:
+        maf = CFG["inputs"]["input_maf"]
+    output:
+        maf = CFG["dirs"]["inputs"] + "master_maf/{seq_type}/input.maf"
+    run:
+        op.absolute_symlink(input.maf, output.maf)
+
+rule _hotmaps_input_subsetting_categories:
+    input:
+        subsetting_categories = ancient(CFG["inputs"]["subsetting_categories"])
+    output:
+        subsetting_categories = CFG["dirs"]["inputs"] + "sample_sets/subsetting_categories.tsv"
+    run:
+        op.absolute_symlink(input.subsetting_categories, output.subsetting_categories)
+
+checkpoint _hotmaps_prep_input:
+    input:
+        maf = expand(
+            str(rules._hotmaps_input_maf.output.maf),
+            seq_type = CFG["samples"]["seq_type"].unique()
+            ),
+        subsetting_categories = ancient(str(rules._hotmaps_input_subsetting_categories.output.subsetting_categories))
+    output:
+        CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/done"
+    log:
+        stdout = CFG["logs"]["inputs"] + "{sample_set}--{launch_date}/prep_input/prep_input_maf.log"
+    conda:
+        CFG["conda_envs"]["prepare_mafs"]
+    params:
+        include_non_coding = str(CFG["maf_processing"]["include_non_coding"]).upper(),
+        mode = "HotMAPS",
+        metadata_cols = CFG["samples"],
+        metadata = CFG["samples"].to_numpy(na_value='')
+    script:
+        PREPARE_MAFS
+
+rule _hotmaps_split_dnps:
+    input:
+        maf = CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/{md5sum}.maf"
+    output:
+        dnps = temp(CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/{md5sum}.dnps.maf"),
+        completed = CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/{md5sum}_split_dnps.complete"
+    shell:
+        op.as_one_line("""
+        variant_type_col=$(head -n 1 {input.maf} | sed 's/\\t/\\n/g' | nl | grep "Variant_Type" | cut -f 1) &&
+        protein_position_col=$(head -n 1 {input.maf} | sed 's/\\t/\\n/g' | nl | grep "Protein_position" | cut -f 1) &&
+        cat <( head -n 1 {input.maf} ) <( awk -v var_col="$variant_type_col" -v protein_col="$protein_position_col" ' {{ if ( $var_col=="DNP" && $protein_col ~ /[0-9?]+[-][0-9?]+/) print $0 }} ' {input.maf} ) > {output.dnps} &&
+        touch {output.completed}
+        """)
+
+checkpoint _hotmaps_maf2vcf:
```

Comment:
I think if the conda env is used only for this it should work with the yaml from vcf2maf module

Reply:
I think you are right -- I will change the vcf2maf conda environment to use the existing one and make a separate one for bcftools and test the module to make sure it works

Reply:
thanks!!

Reply:
I tested the module with the existing conda environment, Initially the rule _hotmaps_vcf2maf would fail using the older vcf2maf version so I had to add an option in the default.yaml config to provide the src dir of a custom vcf2maf script
```
options:
            vcf2maf:
                src_dir: "{REPODIR}/modules/vcf2maf/1.3/src"
```
And add some more code in the rule _hotmaps_vcf2maf to point to the new script. Plus I added some error handling steps because when I ran it initially, some MAFs would return blank and had to be run through vcf2maf again! I also created a different conda environment that just contains the updated bcftools version, and I removed the old vcf2maf environment that I had made from the conda_environments directory in scratch. I tested running the module with all the changes and it works!